charmcraft/commands/build.py

```diff
@@ -138,24 +140,79 @@ def __init__(self, args, config):
         self.ignore_rules = self._load_juju_ignore()
         self.config = config
 
-    def run(self):
-        """Build the charm."""
+    def build_charm(self, bases_config: Optional[BasesConfiguration]) -> str:
+        """Build the charm.
+
+        :param bases_config: Bases configuration to use for build.
+
+        :returns: File name of charm.
+        """
         logger.debug("Building charm in '%s'", self.buildpath)
 
         if self.buildpath.exists():
             shutil.rmtree(str(self.buildpath))
         self.buildpath.mkdir()
 
-        create_manifest(self.buildpath, self.config.project.started_at)
+        create_manifest(self.buildpath, self.config.project.started_at, bases_config)
 
         linked_entrypoint = self.handle_generic_paths()
         self.handle_dispatcher(linked_entrypoint)
         self.handle_dependencies()
-        zipname = self.handle_package()
+        zipname = self.handle_package(bases_config)
 
         logger.info("Created '%s'.", zipname)
         return zipname
 
+    def run(self) -> List[str]:
+        """Run build process.
+
+        In managed-mode (and eventually destructive-mode), build for each bases
```

Comment:
```suggestion
        In managed-mode (and eventually destructive-mode), build for each base
```

Reply:
I have been uses `bases configuration` to imply a unit/entry of `bases:` vs `base configuration` when specifically discussing the content of a `base` object.